Fix git shallow update

robinro opened the pull request:
##### ISSUE TYPE
- Bugfix Pull Request
##### COMPONENT NAME

source_control/git.py
##### ANSIBLE VERSION

devel
##### SUMMARY

Fix #3782, which was introduced by f1bacc1.

Fetch (updating a repo) failed silently when depth is given and version==HEAD.

This needs to be cherry-picked to 2.1-stable too since it break ansible-pull for people not specifying a version, i.e. the minimal example.

Commit:
remove unused variables

Commit:
fetch branch name instead of HEAD

fix #3782, which was introduced by f1bacc1d3f9578f26d4ae2f66112cbb2509a7fe8

gregdek commented:
Thanks @robinro for this PR. This module is maintained by the Ansible core team, so it can take a while for patches to be reviewed. Thanks for your patience.

Core team: please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with 'needs_revision' or merge as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

Commit:
disable git depth option for old git versions

fixes #3782
git support for `--depth` did not fully work in old git versions (before 1.8.2)
fall back to full clones/fetches on those versions

abadger commented:
Merged to devel and stable-2.1 branches.

abadger commented:
@robinro -- Had to revert this because it broke integration tests.  Want to take a look and then resubmit?  (I'm back home from pycon now so feel free to ping me in IRC when it's ready.)
